specification.md

```diff
@@ -0,0 +1,191 @@
+
+#### Extract a `SpanContext` from a carrier
+
+Required parameters
+    - A **format** descriptor (typically but not necessarily a string constant) which tells the `Tracer` implementation how to decode `SpanContext` from the carrier parameter
+    - A **carrier**, whose type is dictated by the **format**. The `Tracer` implementation will decode the `SpanContext` from this carrier object according to **format**.
+
+**Returns** a `SpanContext` instance suitable for use as a **reference** when starting a new `Span` via the `Tracer`.
+
+#### Note: required **format**s for injection and extraction
+
+Both injection and extraction rely on an extensible **format** parameter that dictates the type of the associated "carrier" as well as how a `SpanContext` is encoded in that carrier. All of the following **format**s must be supported by all Tracer implementations.
+    - **Text Map**: an arbitrary string-to-string map with an unrestricted character set for both keys and values
+    - **HTTP Headers**: a string-to-string map with keys and values that are suitable for use in HTTP headers. That is, keys must not require strict casing and values must be URL-encoded
+    - **Binary**: a (single) arbitrary binary blob representing a `SpanContext`
+
+### `Span`
+
+With the exception of the method to retrieve the `Span`'s `SpanContext`, none of the below may be called after the `Span` is finished.
+
+#### Retrieve the `Span`s `SpanContext`
+
+There should be no parameters.
+
+**Returns** the `SpanContext` for the given `Span`
+
+#### Overwrite the operation name
+
+Required parameters
+    - The new **operation name**, which supersedes whatever was passed in when the `Span` was started
+
+#### Set `Span` tags
+
+Required parameters
+    - The tag key, which must be a string
+    - The tag value, which must be either a string, a boolean value, or a numeric type
+
+Note that the OpenTracing project documents certain **"standard tags"** which have prescribed semantic meanings. TODO: add these tags to a central .yaml file and link here.
+
+#### Log structured data
+
+Required parameters
+    - One or more key:value pairs, where the keys must be strings and the values may have any type at all. Some OpenTracing implementations may handle more (or more of) certain log values than others.
+
+Optional parameters
+    - An explicit timestamp. If specified, it must fall between the local start and finish time for the span.
+
+Note that the OpenTracing project documents certain **"standard log keys"** which have prescribed semantic meanings. TODO: add these log keys to a central .yaml file and link here.
+
+#### Set a **baggage** item
+
+Baggage items are key:value string pairs that apply to the given `Span`, its `SpanContext`, and **all `Spans` which directly or transitively _reference_ the local `Span`.** That is, baggage items propagate in-band along with the trace itself.
```

Comment:
I don't think we should make that clarification until other reference types exist.